### PR TITLE
Rename sampleapp variables

### PR DESCRIPTION
**Pull request by pyhys**
Subslots are numbered for example 0x8001, and we have around 5
positions in an array to store subslot data (for each slot).
Typically the variable 'subslot' is used for the subslot number.
Clarify the difference between the subslot number and the index into the
array.
    
Also improved NULL-check

### CODE REVIEW

File: sample_app/sampleapp_common.c [hunk 1]
```diff
@@ -215,27 +215,43 @@ void app_print_network_details (
...
+ *         NULL if subslot is not found/plugged.
  */
 static app_subslot_t * app_subslot_get (
    app_data_t * p_appdata,
    uint16_t slot_nbr,
    uint16_t subslot_nbr)
 {
-   uint16_t subslot;
-   for (subslot = 0; subslot < PNET_MAX_SUBSLOTS; subslot++)
+   uint16_t subslot_array_index; /* Range 0 - (PNET_MAX_SUBSLOTS - 1) */
```
Comment (olbjo) [inline sample_app/sampleapp_common.c:238]: I see that the name "sub_ix" is used at least at one time in the stack. Maybe such a shorter name is to prefer? Or at least shorted to "subslot_index".

Also I think that the range is quite clear given the conditions on the for loop. I do not think the comment adds much.

**Review by olbjo — APPROVED** [review]
As you see by my comments I don't agree on the renaming.
Go ahead and merge if you want to. I do not see any errors. 😄

File: sample_app/sampleapp_common.c [hunk 2]
```diff
@@ -215,27 +215,43 @@ void app_print_network_details (
 
 /************************** Subslot data *************************************/
 
+/* TODO: We do not unplug modules from slots at release or abort.
```
[the remark this answers is not in the document]
Comment (pyhys) [reply sample_app/sampleapp_common.c]: Removed.
Done.

File: sample_app/sampleapp_common.c (hunk 1 above)
Comment (pyhys) [reply sample_app/sampleapp_common.c]: Good point. 
Renamed to `subslot_ix`
Done

**Comment by olbjo** [comment]
Great!

**Pull request merged**